---
title: "Gene expression of L. plantarum WCFS1"
author: "Ivar van den Akker, Herke Schuffel, Yuri Wit en Peter Cserei"
date: "11/24/2020"
output:
  word_document:
    toc: yes
    toc_depth: '4'
  pdf_document:
    toc: yes
    toc_depth: 4
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '4'
toctitle: Inhoudsopgave
---

#### Library importeren

```{r limma edgeR}
library(limma)
library(edgeR)
library(dplyr)
library(xlsx)
library(factoextra)
library(cluster)
library(latexpdf)
library(tinytex)
library(pathview)
```

#### Bestand inladen

```{r}
fDir <- "./"
fName <- "RNA-Seq-counts.txt"
cnts <- read.delim(paste0(fDir,fName), comment.char="#")
row.names(cnts) <- cnts[,"ID"]

f2Name <- "WCFS1_anno.txt"
annotation <- read.delim(paste0(fDir,f2Name), comment.char = "#")
row.names(annotation) <- annotation[,row_number(0)]
```

#### DGE list maken

```{r}
label <- c("WCFS1.glc",  "WCFS1.glc",	"WCFS1.rib",	"WCFS1.rib",	"NC8.glc",	"NC8.glc",	"NC8.rib",	"NC8.rib")
group <- factor(label)
y <- DGEList(counts=cnts[,2:9],group=group)
```

#### Data normaliseren

```{r}
y <- calcNormFactors(y, method="TMM" )
print(y$samples)
```

#### Filteren op low counts

```{r}
keep <- filterByExpr(y)
y <- y[keep,]
```

#### Design matrix maken

```{r}
design <- model.matrix(~0+group, data=y$samples)
colnames(design) <- levels(y$samples$group)
print(design)
```

#### Despersie berekenen

```{r}
y <- estimateDisp(y, design)

f <- estimateGLMCommonDisp(y,design)
f <- estimateGLMTrendedDisp(y,design, method="power")
f <- estimateGLMTagwiseDisp(y,design)
```

#### Fit data

```{r}
fit <- glmFit(y,design)
```

#### Determine fold changes

```{r}
WCFSglcrib <- makeContrasts(exp.r=WCFS1.glc-WCFS1.rib, levels=design)
WCFSglcribfit <- glmLRT(fit, contrast=WCFSglcrib)
NC8glcrib <- makeContrasts(exp.r=NC8.glc-NC8.rib, levels=design)
NC8glcribfit <- glmLRT(fit, contrast=NC8glcrib)
```

#### Print top tags

```{r}
WCFSglcribres<-topTags(WCFSglcribfit, n=nrow(WCFSglcribfit))
```

```{r}
NC8glcribres<-topTags(NC8glcribfit, n=nrow(NC8glcribfit))
```

#### Cluster genes using hierarchal clustering

```{r}
x <- t(y$counts)
x <- dist(x, method = "euclidean")
x <- hclust(x, method = "average")
plot(x)
```

#### PCA plot

```{r}
plotMDS(y)
```

#### Dispersie plot

```{r}
plotBCV(y)
plotBCV(f)
```

#### Filteren

```{r}
WCFSfinal <- filter(WCFSglcribres$table,  WCFSglcribres$table$FDR<0.05 &(WCFSglcribres$table$logFC>1|WCFSglcribres$table$logFC<(-1)))
```


```{r}
NC8final <- filter(NC8glcribres$table,  NC8glcribres$table$FDR<0.05 &(NC8glcribres$table$logFC>1|NC8glcribres$table$logFC<(-1)))
```


```{r}
WCFS1_sig <- cbind(WCFSfinal, annotation[rownames(WCFSfinal),])
NC8_sig <- cbind(NC8final, annotation[rownames(NC8final),])
WCFS1_sorted <- WCFS1_sig[order(WCFS1_sig[, "logFC"], decreasing = TRUE, na.last = FALSE), ,drop=FALSE]
NC8_sorted <- NC8_sig[order(NC8_sig[, "logFC"], decreasing = TRUE, na.last = FALSE), ,drop=FALSE]
```


```{r}
#write.xlsx(WCFS1_sorted, file = "Merged_Genes.xlsx", sheetName = "WCFS1_data", 
#  col.names = TRUE, row.names = TRUE, append = FALSE)
#write.xlsx(NC8_sorted, file = "Merged_Genes.xlsx", sheetName = "NC8_data", 
#  col.names = TRUE, row.names = TRUE, append = TRUE)
```

#### Heat Map
```{r}
heatmap(as.matrix(y), cexCol = 0.9, main="Heatmap counts")
```

```{r}
subWCFS <- subset(WCFSfinal, select = -c(logCPM,LR,PValue,FDR))
subNC8 <- subset(NC8final, select = -c(logCPM,LR,PValue,FDR))

subWCFS <- cbind(id = rownames(subWCFS), subWCFS)
subNC8 <- cbind(id = rownames(subNC8), subNC8)

mergeNC8WCFS <- merge(x = subWCFS, y = subNC8, by = "id", all.x = TRUE)

colnames(mergeNC8WCFS)[2] <- "WCFS1"
colnames(mergeNC8WCFS)[3] <- "NC8"

heatmap(data.matrix(subset(mergeNC8WCFS, select = -c(id))), cexCol = 1, main="Heatmap Foldchanges")
```

#### KEGG Mapper
```{r}
#Define pathway to be used (found on KEGG db)
glycolysispathway <- "lpl00010"
pentosephyosphatepathway <- "lpl00030"
#Load selected genes
WCFS1genes <- read.xlsx("Merged_Genes.xlsx", 1)
NC8genes <- read.xlsx("Merged_Genes.xlsx", 2)
#Seperate logFC of selected genes
WCFS1logFC <- WCFS1genes$logFC
NC8logFC <- NC8genes$logFC
#Set names of the data
names(WCFS1logFC) <- WCFS1genes$NA.
names(NC8logFC) <- NC8genes$NA.
#Generate pathviews
WCFS1.out <- pathview(gene.data = WCFS1logFC, species = "lpl", pathway = glycolysispathway, gene.idtype = "KEGG")
file.copy("lpl00010.pathview.png", "WCFS1.lpl00010.pathview.png", overwrite = TRUE)
NC8.out <- pathview(gene.data = NC8logFC, species = "lpl", pathway = glycolysispathway, gene.idtype = "KEGG")
file.copy("lpl00010.pathview.png", "NC8.lpl00010.pathview.png", overwrite = TRUE)

WCFS1.out <- pathview(gene.data = WCFS1logFC, species = "lpl", pathway = pentosephyosphatepathway, gene.idtype = "KEGG")
file.copy("lpl00030.pathview.png", "WCFS1.lpl00030.pathview.png", overwrite = TRUE)
NC8.out <- pathview(gene.data = NC8logFC, species = "lpl", pathway = pentosephyosphatepathway, gene.idtype = "KEGG")
file.copy("lpl00030.pathview.png", "NC8.lpl00030.pathview.png", overwrite = TRUE)
```

#### K-means
```{r}
km <- kmeans((WCFSfinal %>% select(-(logCPM:FDR))), centers = 2)
fviz_cluster(km, data = (WCFSfinal %>% select(-(LR:FDR))))
km <- kmeans((NC8final %>% select(-(logCPM:FDR))), centers = 2)
fviz_cluster(km, data = (NC8final %>% select(-(LR:FDR))))
```
